pixeldata/src/gdcm.rs

```diff
@@ -144,6 +145,164 @@ where
+                let gdcm_error_mapper = |source: GDCMError| DecodeError::Custom {
+                    message: source.to_string(),
+                    source: Some(Box::new(source)),
+                };
+
+                let frame = frame as usize;
+                let data = if number_of_frames == 1 && fragments.len() > 1 {
+                    let mut buf = Vec::new();
+                    fragments
+                        .into_iter()
+                        .for_each(|fragment| buf.append(&mut fragment.clone()));
+                    buf
+                } else {
+                    fragments[frame].to_vec()
+                };
+
+                let frame = match ts_type {
+                    GDCMTransferSyntax::ImplicitVRLittleEndian
+                    | GDCMTransferSyntax::ExplicitVRLittleEndian => {
+                        let frame_size = cols * rows * samples_per_pixel * (bits_allocated / 8);
+                        data.chunks_exact(frame_size as usize)
+                            .nth(frame)
+                            .map(|frame| frame.to_vec())
+                            .unwrap_or_default()
+                    }
+                    _ => {
+                        let buffer = [data.as_slice()];
+                        let dims = [cols.into(), rows.into(), 1];
+
+                        decode_multi_frame_compressed(
+                            &buffer,
+                            &dims,
+                            pi_type,
+                            ts_type,
+                            samples_per_pixel,
+                            bits_allocated,
+                            bits_stored,
+                            high_bit,
+                            pixel_representation as u16,
+                        )
+                        .map_err(gdcm_error_mapper)
+                        .context(DecodePixelDataSnafu)?
+                        .to_vec()
+                    }
+                };
+
+                if planar_configuration == 1 && samples_per_pixel == 3 {
+                    let frame_size =
+                        (cols as usize * rows as usize * (bits_allocated as usize / 8)) as usize;
+                    let mut interleaved = Vec::with_capacity(frame.len());
+
+                    for i in 0..frame_size {
+                        interleaved.push(frame[i]);
+                        interleaved.push(frame[i + frame_size]);
+                        interleaved.push(frame[i + frame_size * 2]);
+                    }
+                    interleaved
+                } else {
+                    frame
+                }
+            }
+            Value::Primitive(p) => {
+                // Non-encoded, just return the pixel data of the first frame
+                p.to_bytes().to_vec()
+            }
+            Value::Sequence(_) => InvalidPixelDataSnafu.fail()?,
+        };
+
+        // pixels are already interpreted,
+        // set new photometric interpretation
```

Comment:
We will still need this, because the decoded pixel data does not necessarily follow the photometric interpretation originally described.